### PR TITLE
Align price field loading with other forms to address notice

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Align price field loading with other forms to address notice

Before
----------------------------------------
Price fields loaded both the legacy way & the way that is consistent with other forms. Notice on old way, hooks called twice

After
----------------------------------------
I also moved the related  financialacl hook code to the the extension - the output is kinda messed up before & after
Before

image

After
image

After in non-pop up mode 
image

but I don't think you could make the argument that we should not move the code to the extension based on before being so much better

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Event/Form/Participant.php [hunk 1]
```diff
@@ -2071,15 +2084,11 @@ public function getInvoiceID(): string {
    * change again.
    */
   private function buildAmount(): void {
-    $feeFields = $this->_values['fee'] ?? NULL;
-    //its time to call the hook.
-    CRM_Utils_Hook::buildAmount('event', $this, $feeFields);
```
Comment (eileenmcnaughton) [inline CRM/Event/Form/Participant.php:2076]: This is already called in the `getPriceFieldMetaData()` call

**Comment by demeritcowboy** [comment]
There's no tax error but now there's no money either, and a new error:

On contact event tab:

untitled5

On viewing the participant:

untitled6

**Comment by eileenmcnaughton** [comment]
Great - we fixed inflation!

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by demeritcowboy** [comment]
So far so good. It's obviously impossible to run everything here so I think I'm good to merge.

I have a memory of it automatically setting the participant status to registered when you complete a pending, but it seems a pre-existing problem. It gets the numbers right anyway. And it does update it when you complete a partially paid.

**Comment by eileenmcnaughton** [comment]
thanks @demeritcowboy

**Pull request merged**